pennylane_lightning/core/lightning_newAPI_base.py

```diff
@@ -0,0 +1,348 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+This module contains the :class:`~.LightningKokkos` class, a PennyLane simulator device that
+interfaces with C++ for fast linear algebra calculations.
+"""
+from numbers import Number
+from typing import Callable, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import pennylane as qml
+from pennylane.devices import DefaultExecutionConfig, Device, ExecutionConfig
+from pennylane.devices.modifiers import simulator_tracking, single_tape_support
+from pennylane.tape import QuantumScript, QuantumTape
+from pennylane.typing import Result, ResultBatch
+
+Result_or_ResultBatch = Union[Result, ResultBatch]
+QuantumTapeBatch = Sequence[QuantumTape]
+QuantumTape_or_Batch = Union[QuantumTape, QuantumTapeBatch]
+PostprocessingFn = Callable[[ResultBatch], Result_or_ResultBatch]
+
+
+@simulator_tracking
+@single_tape_support
+class LightningBase(Device):
+    """PennyLane Lightning Kokkos device.
+
+    A device that interfaces with C++ to perform fast linear algebra calculations.
+
+    Use of this device requires pre-built binaries or compilation from source. Check out the
+    :doc:`/lightning_kokkos/installation` guide for more details.
+
+    Args:
+        wires (int): the number of wires to initialize the device with
+        sync (bool): immediately sync with host-sv after applying operations
+        c_dtype: Datatypes for statevector representation. Must be one of
+            ``np.complex64`` or ``np.complex128``.
+        shots (int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to ``None`` if not specified. Setting
+            to ``None`` results in computing statistics like expectation values and
+            variances analytically.
+        kokkos_args (InitializationSettings): binding for Kokkos::InitializationSettings
+            (threading parameters).
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    # General device options
+    _new_API = True
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        wires,
+        *,
+        c_dtype=np.complex128,
+        shots=None,
+        batch_obs=False,
+    ):
+        super().__init__(wires=wires, shots=shots)
+
+        self._c_dtype = c_dtype
+        self._batch_obs = batch_obs
+
+        if isinstance(wires, int):
+            self._wire_map = None  # should just use wires as is
+        else:
+            self._wire_map = {w: i for i, w in enumerate(self.wires)}
+
+        # Dummy for LightningStateVector, LightningMeasurements, LightningAdjointJacobian
+        self.LightningStateVector: Callable = None
+        self.LightningMeasurements: Callable = None
+        self.LightningAdjointJacobian: Callable = None
```

Comment:
The base class doesn't need to know about these callables. Could we remove them from the base class?

Reply:
Why do you suggest removing it? :thinking:

Reply:
I'd like to generalize the base class as much as possible. I wonder if these dummy callables can be abstracted away in anyways. But, I think we need them right now to preserve consistency between the method signature in base and concrete device classes.